test_scripts/API/RAI/049_ATF_N_steeringWheelLocation_IsWrongDataType_WARNINGS_RAI_steeringWheelLocation.lua

```diff
@@ -0,0 +1,265 @@
+-- [MOBILE_API] [HMI_API] The 'steeringWheelLocation' enum
+-- [MOBILE_API] The 'steeringWheelLocation' param
+-- [HMI RPC validation]: SDL must send log error and ignore invalid RPC in case SDL cuts off fake parameters and RPC becomes invalid
+-- [RegisterAppInterface] WARNINGS appHMIType(s) partially coincide or not coincide with current 
+-- non-empty data stored in PolicyTable
+--
+-- Description:
+-- In case SDL does NOT receive value of "steeringWeelLocation" parameter via UI.GetCapabilities_response from HMI
+-- SDL must retrieve the value of "steeringWeelLocation" parameter from "HMI_capabilities.json" file and 
+-- provide the value of "steeringWeelLocation" via RegisterAppInterface_response to mobile app
+--
+-- 1. Used preconditions
+-- In InitHMI_OnReady HMI replies with parameter steeringWeelLocation with wrong data type to UI.GetCapabilities
+-- SDL invalidates message
+-- Get value of "steeringWeelLocation" parameter from "HMI_capabilities.json".
+--
+-- 2. Performed steps
+-- Register new applications with conditions for result WARNINGS
+--
+-- Expected result:
+-- SDL->mobile: RegisterAppInterface_response(WARNINGS, success: true) 
+-- steeringWeelLocation is equal to "HMI_capabilities.json"
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local testCasesForRAI = require('user_modules/shared_testcases/testCasesForRAI')
+local events = require("events")
+local mobile_session = require('mobile_session')
+
+--[[ Local variables ]]
+local value_steering_wheel_location = testCasesForRAI.get_data_steeringWheelLocation()
+
+--[[ Local functions ]]
+
+--[[@update_sdl_preloaded_pt_json - update preloaded_pt
+--! @ in case REAI to return response(WARNINGS, success:true)
+--! @parameters: NO
+--]]
+local function update_sdl_preloaded_pt_json()
+	local pathToFile = config.pathToSDL .. 'sdl_preloaded_pt.json'
+	local file = io.open(pathToFile, "r")
+	local json_data = file:read("*all") -- may be abbreviated to "*a";
+	file:close()
+				
+	local json = require("modules/json")
+	
+	local data = json.decode(json_data)
+	for k in pairs(data.policy_table.functional_groupings) do
+		if (data.policy_table.functional_groupings[k].rpcs == nil) then
+			data.policy_table.functional_groupings[k] = nil
+		end
+	end
+
+	data.policy_table.app_policies["0000001"] = {
+		keep_context = false,
+		steal_focus = false,
+		priority = "NONE",
+		default_hmi = "NONE",
+		groups = {"Base-4"}
+	}
+	data.policy_table.app_policies["0000001"].AppHMIType = {"NAVIGATION"}
```

Comment:
@istoimenova
I propose move AppHMIType parameter initialization to structure above:
```
	data.policy_table.app_policies["0000001"] = {
 		keep_context = false,
 		steal_focus = false,
 		priority = "NONE",
 		default_hmi = "NONE",
 		groups = {"Base-4"},
		AppHMIType = {"NAVIGATION"}
 	}
```